### PR TITLE
[SC-914] fix consent emails being sent to random users

### PR DESCRIPTION
**Pull request by Parnswir**

### CODE REVIEW

File: controllers/administration.js [hunk 1]
```diff
@@ -1164,31 +1164,35 @@ const getUsersWithoutConsent = async (req, roleName, classId) => {
...
-
-    const usersWithoutConsent = users.filter(user => {
-        let userWithConsent = consents.find(consent => {
-            return consent.userId === user._id;
-        });
+    const consents = (await api(req).get('/consents', {
+        qs: {
+            userId: { $in: users.map(u => u._id) },
+            $populate: 'userId',
+            $limit: false,
```
Comment (CeEv) [inline controllers/administration.js]: Can you test if $limit=false send from client work? I expect that our global server hooks format the boolean to a string and therefore it do not work.

**Comment by CeEv** [comment]
The name and logic in getUsersWithoutConsent sounds like a top level route for the api, that can call from client.

**Comment by janrenz** [comment]
It sounds like this should be not a own Route in the server but maybe a filter

File: controllers/administration.js (hunk 1 above)
Comment (Parnswir) [reply controllers/administration.js]: **without $limit**

limit
-----------------------
**$limit: false**

limit_false

**Comment by Parnswir** [comment]
Let's move the discussion of how and when to here: https://ticketsystem.schul-cloud.org/browse/SC-945

**Pull request merged**